packages/strapi-hook-bookshelf/lib/buildQuery.js

```diff
@@ -53,6 +31,165 @@ const buildQuery = ({ model, filters }) => qb => {
   }
 };
 
+/**
+ * Add joins and where filters
+ * @param {Object} qb - knex query builder
+ * @param {Object} model - Bookshelf model
+ * @param {Array<Object>} whereClauses - an array of where clause
+ */
+const buildJoinsAndFilter = (qb, model, whereClauses) => {
+  const aliasMap = {};
+  /**
+   * Returns an alias for a name (simple incremental alias name)
+   * @param {string} name - name to alias
+   */
+  const generateAlias = name => {
+    if (!aliasMap[name]) {
+      aliasMap[name] = 1;
+    }
+
+    return `${name}_${aliasMap[name]++}`;
+  };
+
+  /**
+   * Build a query joins and where clauses from a query tree
+   * @param {Object} qb - Knex query builder
+   * @param {Object} tree - Query tree
+   */
+  const buildQueryFromTree = (qb, queryTree) => {
+    // build joins
+    Object.keys(queryTree.children).forEach(key => {
+      const subQueryTree = queryTree.children[key];
+      buildJoin(qb, subQueryTree.assoc, queryTree, subQueryTree);
+
+      buildQueryFromTree(qb, subQueryTree);
+    });
+
+    // build where clauses
+    queryTree.where.forEach(w => buildWhereClause({ qb, ...w }));
+  };
+
+  /**
+   * Add table joins
+   * @param {Object} qb - Knex query builder
+   * @param {Object} assoc - Models association info
+   * @param {Object} originInfo - origin from which you are making a join
+   * @param {Object} destinationInfo - destination with which we are making a join
+   */
+  const buildJoin = (qb, assoc, originInfo, destinationInfo) => {
+    if (assoc.nature === 'manyToMany') {
```

Comment:
There is no error in this snippet of code but it seems that we haven't the same linter rules in the backend and frontend parts...

The frontend linter will ask to use this pattern
```js

if (condition) {
  // Do stuff.
  return;
}

// Do other stuff
```

instead of
```js
if (condition) {
  // Do stuff
} else {
  // Do other stuff
}
```